activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb

```diff
@@ -17,13 +17,10 @@ def select_all(arel, name = nil, binds = [], preparable: nil)
         # as values.
         def select_one(arel, name = nil, binds = [])
           arel, binds = binds_from_relation(arel, binds)
-          @connection.query_options.merge!(as: :hash)
```

Comment:
Actually, it's a bug fix. But I cannot reliably reproduce the bug. Sometimes '#select_one` would return an array instead of a hash, but with this patch, I get a hash every time.